blog-content/supporting-opentelemetry-metrics.md

```diff
@@ -0,0 +1,333 @@
+ClickHouse. While we use the OpenTelemetry collector to receive, deserialize, and batch data, we export to our Golang
+API that mutates the data before writing to Apache Kafka. A set of workers (the Apache Kafka Connect ClickHouse
+exporter) read the data and write it to ClickHouse in large batches.
+
+```text
+            +------------------+
+            |  OTel Collector  |
+            +------------------+
+                    |
+                    v
+            +------------------+
+            |   Highlight API  |
+            +------------------+
+                    |
+                    v
+            +------------------+
+            |   Apache Kafka   |
+            +------------------+
+                    |
+         +----------------------+
+         |     Kafka Connect    |
+         +----------------------+
+                    |
+                    v
+           +------------------+
+           |    ClickHouse    |
+           +------------------+
+```
+
+### OpenTelemetry Collector Setup
+
+The OpenTelemetry Collector is a key component in an OTel pipeline, responsible for receiving, processing, and exporting
+telemetry data. For metric ingestion into ClickHouse, we configure the collector to receive OTel metrics via the OTLP
+receiver, process them using built-in processors (e.g., batch and transform), and export them to our API.
+
+Here’s an example OpenTelemetry Collector configuration for exporting metrics to our Highlight API which then batch
+exports data to ClickHouse:
+
+```yaml
+receivers:
+  awsfirehose/cwmetrics:
+    record_type: cwmetrics
+  awsfirehose/otlp_v1:
+    record_type: otlp_v1
+  otlp:
+    protocols:
+      grpc:
+      http:
+
+processors:
+  batch:
+    timeout: 5s
+    send_batch_size: 1000
+
+exporters:
+  otlphttp:
+    endpoint: 'http://pub.prod.vpc.highlight.io:8082/otel'
+
+service:
+  pipelines:
+    metrics:
+      receivers: [ otlp, awsfirehose/cwmetrics, awsfirehose/otlp_v1 ]
+      processors: [ batch ]
+      exporters: [ otlphttp ]
+```
+
+See our full production OpenTelemetry
+collector [config here](https://github.com/highlight/highlight/blob/bb84dab01aab4ec223632814b244e3381e074b26/deploy/otel-collector.yaml).
+
+If you are building an OpenTelemetry pipeline from scratch, you can use the `clickhouse` collector export for direct
+writes to the database. For our production use-case, we route the data through our API for pre-processing and write
+buffering via Apache Kafka, but you may find success with the exporter even for large volumes.
+
+```yaml
+exporters:
+  clickhouse:
+    endpoint: "tcp://clickhouse-server:9000"
+    database: "otel_metrics"
+    username: "default"
+    password: ""
+```
+
+By using the OpenTelemetry collector as the initial entrypoint for the data, we get the benefit of the collector supporting myriad receivers which can be compatible with different data formats.
```

Comment:
```suggestion
By using the OpenTelemetry collector as the initial entrypoint for the data, we get the benefit of the collector supporting a myriad of receivers which can be compatible with different data formats.
```

Reply:
https://www.writersdigest.com/write-better-fiction/myriad-vs-myriad-of-grammar-rules